rfcs/0000-graphql-api.md

```diff
@@ -0,0 +1,433 @@
+}
+
+type SnarkFeeUpdate {
+  fee: SnarkFee
+  consensus: ConsensusState!
+}
+
+type Block {
+  coinbase: UInt32!
+  creator: PublicKey!
+  payments: [Payment]!
+  snarkFees: [SnarkFee]!
+}
+
+type BlockUpdate {
+  block: Block!
+  consensus: ConsensusState!
+}
+
+type Balance {
+  total: UInt64!,
+  unknown: UInt64!,
+}
+
+type Wallet {
+  publicKey: PublicKey!
+  balance: Balance!
+}
+
+type NodeStatus {
+  network: String
+}
+
+## Input types
+
+input AddWalletInput {
+  public: PublicKey
+  private: PrivateKey
+}
+
+input DeleteWalletInput {
+  public: PublicKey
+}
+
+input AddPaymentReceiptInput {
+  receipt: String
+}
+
+input SetNetworkInput {
+  address: String
+}
+
+input SetSnarkWorkerInput {
+  worker: PublicKey!
+  fee: UInt32!
+}
+
+input CreatePaymentInput {
+  from: PublicKey!,
+  to: PublicKey!,
+  amount: UInt64!,
+  fee: UInt32!,
+  memo: String
+}
+
+input PaymentFilterInput {
+  toOrFrom: PublicKey,
+}
+
+input BlockFilterInput {
+  creator: PublicKey,
+}
+
+## Payload types
+
+type CreatePaymentPayload {
+  payment: Payment
+}
+
+type SetSnarkWorkerPayload {
+  worker: SnarkWorker
+}
+
+type SetNetworkPayload {
+  address: String
+}
+
+type AddPaymentReceiptPayload {
+  payment: Payment
+}
+
+type AddWalletPayload {
+  publicKey: PublicKey
+}
+
+type DeleteWalletPayload {
+  publicKey: PublicKey
+}
+
+# Pagination types
+
+type PageInfo {
+  hasPreviousPage: Boolean!
+  hasNextPage: Boolean!
+}
+
+type PaymentEdge {
+  cursor: String
+  node: PaymentUpdate
+}
+
+type PaymentConnection {
+  edges: [PaymentEdge]
+  nodes: [PaymentUpdate]
+  pageInfo: PageInfo!
+  totalCount: Int
+}
+
+type BlockEdge {
+  cursor: String
+  node: BlockUpdate
+}
+
+type BlockConnection {
+  edges: [BlockEdge]
+  nodes: [BlockUpdate]
+  pageInfo: PageInfo!
+  totalCount: Int
+}
+
+type Query {
+  # List of wallets for which the node knows the private key
+  ownedWallets: [Wallet!]!
+  
+  wallet(publicKey: PublicKey!): Wallet
+  
+  payments(
+    filter: PaymentFilterInput,
+    first: Int,
+    after: String,
+    last: Int,
+    before: String): PaymentConnection
+  
+  blocks(
+    filter: BlockFilterInput,
+    first: Int,
+    after: String,
+    last: Int,
+    before: String): BlockConnection
+  
+  # Null if node isn't performing snark work
+  currentSnarkWorker: SnarkWorker
+
+  # Current sync status of the node
+  syncState: SyncUpdate!
+  
+  # version of the node (commit hash or version #)
+  version: String!
+  
+  # Network that the node is connected to
+  network: String
+  status: NodeStatus
+}
+
+type Mutation {
+  createPayment(input: CreatePaymentInput!): CreatePaymentPayload
+  
+  setSnarkWorker(input: SetSnarkWorkerInput!): SetSnarkWorkerPayload
+  
+  # Configure which network your node is connected to
+  setNetwork(input: SetNetworkInput!): SetNetworkPayload
+  
+  # Adds transaction to the node (note: Not sure how we want to represent this yet)
+  addPaymentReceipt(input: AddPaymentReceiptInput!): AddPaymentReceiptPayload
```

Comment:
I think we should also add the previous receipt_chain_hash as a parameter since we need to know who is the parent of a payment.